pfs_middleware/pfs_middleware/middleware.py

```diff
@@ -1635,8 +1721,22 @@ def post_object(self, ctx):
             else:
                 raise
 
+        # We need to unmung and then mung the headers because the POST may
+        # supply a value S3API_ETAG_HEADER or LISTING_ETAG_OVERRIDE_HEADER
+        # (or ORIGINAL_MD5_HEADER?).  While merge_object_metadata() will
+        # not let these overwrite an existing header, they can be added to
+        # an object that doesn't have such a header.
+        #
+        # Note that if an ETag header is stale (because NumWrites has
+        # changed) then unmung_etags() will strip it and it will not be
+        # put back by mung_etags().
```

Comment:
Yeah, that should be fine. Since the underlying data changed, we have no use for it any more.